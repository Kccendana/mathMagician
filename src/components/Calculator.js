import React from 'react';
import Button from './Button';

const Calculator = () => (
  <div className="calc-grid">
    <div className="output">
      <div className="prev-operand" />
      <div className="curr-operand" />
    </div>
    <Button label="AC" />
    <Button label="+/-" />
    <Button label="%" />
    <Button label="รท" className="bgColor" />
    <Button label="7" />
    <Button label="8" />
    <Button label="9" />
    <Button label="x" className="bgColor" />
    <Button label="4" />
    <Button label="5" />
    <Button label="6" />
    <Button label="-" className="bgColor" />
    <Button label="1" />
    <Button label="2" />
    <Button label="3" />
    <Button label="+" className="bgColor" />
    <Button label="0" className="span-2" />
    <Button label="." />
    <Button label="=" className="bgColor" />
  </div>

);

export default Calculator;
