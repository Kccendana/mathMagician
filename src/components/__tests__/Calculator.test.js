import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Calculator from "../calculator";

describe('Calculator component', () => {
    it('should match the snapshot', () => {
        const {container } = render(<Calculator />);
        expect(container).toMatchSnapshot();
    });
})