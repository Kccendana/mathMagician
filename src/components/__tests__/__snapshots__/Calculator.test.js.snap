// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component should match the snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h1
      class="title"
    >
      Let us do some calculations!!
    </h1>
    <div
      class="calc-grid"
    >
      <div
        class="output"
      >
        <div
          class="curr-operand"
        >
          0
        </div>
      </div>
      <button
        class="blue"
        type="button"
      >
        AC
      </button>
      <button
        class="blue"
        type="button"
      >
        +/-
      </button>
      <button
        class="blue"
        type="button"
      >
        %
      </button>
      <button
        class="bgColor"
        type="button"
      >
        รท
      </button>
      <button
        class="blue"
        type="button"
      >
        7
      </button>
      <button
        class="blue"
        type="button"
      >
        8
      </button>
      <button
        class="blue"
        type="button"
      >
        9
      </button>
      <button
        class="bgColor"
        type="button"
      >
        x
      </button>
      <button
        class="blue"
        type="button"
      >
        4
      </button>
      <button
        class="blue"
        type="button"
      >
        5
      </button>
      <button
        class="blue"
        type="button"
      >
        6
      </button>
      <button
        class="bgColor"
        type="button"
      >
        -
      </button>
      <button
        class="blue"
        type="button"
      >
        1
      </button>
      <button
        class="blue"
        type="button"
      >
        2
      </button>
      <button
        class="blue"
        type="button"
      >
        3
      </button>
      <button
        class="bgColor"
        type="button"
      >
        +
      </button>
      <button
        class="span-2"
        type="button"
      >
        0
      </button>
      <button
        class="blue"
        type="button"
      >
        .
      </button>
      <button
        class="bgColor"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
